cmd/empire/factories.go

```diff
@@ -269,7 +269,9 @@ func newHBReporter(key, env string) (reporter.Reporter, error) {
 
 func newAuthProvider(c *cli.Context) (dockerauth.AuthProvider, error) {
 	provider := dockerauth.NewMultiAuthProvider()
-	provider.AddProvider(dockerauth.NewECRAuthProvider(ecr.New(newConfigProvider(c))))
+	provider.AddProvider(dockerauth.NewECRAuthProvider(func(region string) dockerauth.ECR {
+		return ecr.New(newConfigProvider(c), &aws.Config{Region: aws.String(region)})
```

Comment:
You could also just close over a map so that the ecr clients are memoized. Something like:

```go
var clients map[region]*ecr.ECR
provider.AddProvider(dockerauth.NewECRAuthProvider(func(region string) dockerauth.ECR {
        var client *ecr.ECR
        client, ok := clients[region]
        if !ok {
                client = ecr.New(newConfigProvider(c), &aws.Config{Region: aws.String(region)})
                clients[region] = client
        }
        return client
}
```

Although, it's probably fine to instantiate a new client every time.